src/libs/ReportUtils.ts

```diff
@@ -1540,38 +1355,30 @@ function getTransactionsWithReceipts(iouReportID) {
  * all requests are receipts that are being SmartScanned. As soon as we have a non-receipt request,
  * or as soon as one receipt request is done scanning, we have at least one
  * "ready" money request, and we remove this indicator to show the partial report total.
- *
- * @param {Object|null} iouReportID
- * @param {Object|null} reportPreviewAction the preview action associated with the IOU report
- * @returns {Boolean}
  */
-function areAllRequestsBeingSmartScanned(iouReportID, reportPreviewAction) {
+function areAllRequestsBeingSmartScanned(iouReportID: string | undefined, reportPreviewAction: OnyxEntry<ReportAction>): boolean {
     const transactionsWithReceipts = getTransactionsWithReceipts(iouReportID);
     // If we have more requests than requests with receipts, we have some manual requests
     if (ReportActionsUtils.getNumberOfMoneyRequests(reportPreviewAction) > transactionsWithReceipts.length) {
         return false;
     }
-    return _.all(transactionsWithReceipts, (transaction) => TransactionUtils.isReceiptBeingScanned(transaction));
+    return transactionsWithReceipts.every((transaction) => TransactionUtils.isReceiptBeingScanned(transaction));
 }
 
 /**
  * Check if any of the transactions in the report has required missing fields
  *
- * @param {Object|null} iouReportID
- * @returns {Boolean}
+ * @param iouReportID
  */
-function hasMissingSmartscanFields(iouReportID) {
+function hasMissingSmartscanFields(iouReportID?: string) {
```

Comment:
In multiple functions `string | undefined` is used instead of optional argument, what do you think to make it more consistent?

Reply:
In this case in my opinion it should be string | undefined and will change that , because we always want to pass a parameter to function but it can be an undefined but there are places where ? make more sense because we dont need to pass the parameter 
for more context I recomend this article https://tkdodo.eu/blog/optional-vs-undefined 😄